app/controllers/admin/bulk_assign_role_controller.rb

```diff
@@ -0,0 +1,70 @@
+module Admin
+  class BulkAssignRoleController < Admin::ApplicationController
+    layout "admin"
+    include Admin::UsersHelper
+
+    def assign_role
+      if permitted_params[:role].blank?
+        raise ArgumentError,
+              I18n.t("admin.bulk_assign_role_controller.role_blank")
+      end
+
+      role = permitted_params[:role]
+      usernames = permitted_params[:usernames].downcase.split(/\s*,\s*/)
+      note = permitted_params[:note_for_current_role].presence
+      note ||= I18n.t("admin.bulk_assign_role_controller.congrats", role: role)
```

Comment:
Thank you for changing the note @rt4914. Not a blocking change, but maybe we could rename the key as well to something more descriptive like `role_assigment` or something similar. Congrats seems very vague.